### PR TITLE
Fixing login crash

### PR DESCRIPTION
**Pull request by JohnOberhauser**
Fixing a crash when a user adds `http://` or `https://` to their server's domain

### CODE REVIEW

File: core/usecase/mastodon/src/main/java/org/mozilla/social/core/usecase/mastodon/auth/Login.kt [hunk 1]
```diff
@@ -104,6 +106,17 @@ class Login(
         clientSecret = null
     }
 
+    private fun extractHost(domain: String): String {
+        return buildString {
```
Comment (devotaaabel) [inline core/usecase/mastodon/src/main/java/org/mozilla/social/core/usecase/mastodon/auth/Login.kt:110]: Can you use `Uri.parse(domain).host` here instead?

Comment (JohnOberhauser) [reply core/usecase/mastodon/src/main/java/org/mozilla/social/core/usecase/mastodon/auth/Login.kt:110]: That will return null if there is no `http://` or `https://`.  That's why I need to do the `buildString` to make sure the url starts with `https://`

Comment (devotaaabel) [reply core/usecase/mastodon/src/main/java/org/mozilla/social/core/usecase/mastodon/auth/Login.kt:110]: Can you try `Uri.fromParts`? I'm not 100% sure as I haven't used it before, but it looks like what you need

Comment (JohnOberhauser) [reply core/usecase/mastodon/src/main/java/org/mozilla/social/core/usecase/mastodon/auth/Login.kt:110]: Hmm, I tried but couldn't figure out a way to do it.

**Review by devotaaabel — APPROVED** [review]

**Pull request merged**